src/Fisher_Kolmogorov_solver.hpp

```diff
@@ -0,0 +1,424 @@
+#ifndef FISHER_KOLMOGOROV_HPP
+#define FISHER_KOLMOGOROV_HPP
+
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/utilities.h>
+
+#include <deal.II/distributed/fully_distributed_tria.h>
+#include <deal.II/base/convergence_table.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_simplex_p.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/fe_values_extractors.h>
+#include <deal.II/fe/mapping_fe.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/fe_values_extractors.h>
+#include <deal.II/fe/mapping_fe.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_in.h>
+
+
+#include <deal.II/grid/grid_in.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/distributed/grid_refinement.h>
+
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/trilinos_precondition.h>
+#include <deal.II/lac/trilinos_sparse_matrix.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <deal.II/base/parameter_handler.h>
+
+#include <fstream>
+#include <iostream>
+
+using namespace dealii;
+
+// Class representing the non-linear diffusion problem.
+template <int dim>
+class FisherKol
+{
+public:
+
+  // Function of the fiber field
+  class FunctionN
+  {
+  public:
+
+    FunctionN(const FisherKol<dim> &fisher_kol) : fisher_kol(fisher_kol) {}
+
+    Tensor<2, dim>
+    value(const Point<dim> & /*p*/) const
+    {
+      Tensor<2, dim> values;
+      Tensor<1, dim> n_vector;
+
+      const double d_ext = fisher_kol.parameters.get_double("coef_dext");
+      const double d_axn = fisher_kol.parameters.get_double("coef_daxn");
+
+      std::string list_string = fisher_kol.parameters.get("normal_vector");
+      std::stringstream ss(list_string);
+      std::string item;
+
+      unsigned int index = 0;
+      while (std::getline(ss, item, ','))
+      {
+        n_vector[index] = std::stod(item);
+        index++;
+      } 
+
+      // making sure the vector is normalized
+      const double norm = n_vector.norm();
+      if (norm != 0.0) n_vector /= norm;
+
+      // computing the D_matrix
+      Tensor<2, dim> identity_matrix = unit_symmetric_tensor<dim>();
+      values = d_ext * identity_matrix + d_axn * outer_product(n_vector, n_vector);
```

Comment:
This is a bit of an ad-hoc optimization, but it's a useful trick to keep in mind.

The value that you return here is always the same (since the diffusion coefficients are constant and so is the axonal direction). Therefore, it is way more efficient to compute this once and for all in the constructor, and then simply return the precomputed value.